core/src/contract_env.rs

```diff
@@ -101,4 +104,15 @@ impl ContractEnv {
         let backend = self.backend.borrow();
         backend.emit_event(&event.to_bytes().unwrap().into())
     }
+
+    pub fn verify_signature(
+        &self,
+        message: &Bytes,
+        signature: &Bytes,
+        public_key: &PublicKey
+    ) -> bool {
+        let (signature, _) = casper_types::crypto::Signature::from_bytes(signature.as_slice())
+            .unwrap_or_else(|_| self.revert(ExecutionError(CouldntDeserializeSignature)));
```

Comment:
```suggestion
            .unwrap_or_else(|_| self.revert(CouldntDeserializeSignature));
```
revert() accepts Into<OdraError> so we don't need to wrap the error